profile/benchmarking/simple.rb

```diff
@@ -164,20 +155,19 @@ def index_document_large(opts={})
           client.create(index: INDEX, body: document)
         end
 
-        with_cleanup do
-          start = current_time
-          results = measured_repetitions.times.collect do
-            Benchmark.realtime do
-              action_iterations.times do
-                client.create(index: INDEX, body: document)
+        duration = with_cleanup do
+          Benchmark.realtime do
+            results = measured_repetitions.times.collect do
+              Benchmark.realtime do
+                action_iterations.times do
```

Comment:
This chunk seems to be repeated quite a bit, i.e.

```ruby
          Benchmark.realtime do
            results = measured_repetitions.times.collect do
              Benchmark.realtime do
                action_iterations.times do
```

Might be worth extracting it to a separate method that would then just be called with a block, putting a highlight on the code under benchmark and not the benchmarking code around it. 

Would also make future changes to the benchmarking structure (like the one in this PR) more localized to that method and not to so many places.

It's probably not possible for all cases as some are a bit different but I think it should be possible for a majority of cases. 

What do you think? :slightly_smiling_face:

Reply:
Good idea, I'll see what I can do. Thanks!